pkg/policy/condition/key.go

```diff
@@ -76,6 +76,15 @@ const (
 
 	// AWSEpochTime - key representing the current epoch time.
 	AWSEpochTime = "aws:EpochTime"
+
+	// AWSPrincipalType - user principal type currently supported values are "User" and "Anonymous".
+	AWSPrincipalType = "aws:principaltype"
+
+	// AWSUserID - user unique ID, in Minio this value is same as your user Access Key.
+	AWSUserID = "aws:userid"
+
+	// AWSUsername - user friendly name, in Minio this value is same as your user Access Key.
+	AWSUsername = "aws:username"
```

Comment:
There is and existing bug in current `Key` enum type that other than `S3XAmzCopySource`, rest are string constants, not enum values.  Its required to declare each enum value for `Key` e.g. `AWSUsername Key = "aws:username"` is correct.